### PR TITLE
Move from ubuntu 18.04 to ubuntu 20.04 in workflows

### PR DESCRIPTION
**Pull request by Czaki**
# Description

Change test on oldest Linux from ubuntu 18.04 to ubuntu 20.04 

## Type of change

- [x] Maintanence

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change
- [ ] example: I check if my changes works with both PySide and PyQt backends
      as there are small differences between the two Qt bindings.  

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/developers/translations.html).

### CODE REVIEW

File: .github/workflows/make_bundle.yml [hunk 1]
```diff
@@ -70,6 +70,7 @@ jobs:
           sudo apt-get install -y libdbus-1-3 libxkbcommon-x11-0 libxcb-icccm4 \
             libxcb-image0 libxcb-keysyms1 libxcb-randr0 libxcb-render-util0 \
             libxcb-xinerama0 libxcb-xfixes0 libxcb-shape0 libqt5gui5
+          sudo ln -s /usr/lib/x86_64-linux-gnu/libffi.so.7/usr/lib/x86_64-linux-gnu/libffi.so.6
```
Comment (andy-sweet) [inline .github/workflows/make_bundle.yml]: Shouldn't there be a space between the source (*so.7) and target (*.so.6)?

More generally, I'm a bit concerned that something is specifically dependent on `/usr/lib/x86_64-linux-gnu/libffi.so.6` and not just `/usr/lib/x86_64-linux-gnu/libffi.so`, but I don't know much about what `bundle.py` does.

**Comment by andy-sweet** [comment]
For reference, here's the official issue on the deprecation of this runner image: https://github.com/actions/runner-images/issues/6002

I'm a bit surprised that Ubuntu 18.04 is being dropped already - there aren't any other images we could use right?

If not, thanks for the quick fix - I just had one comment about the need for a symlink in the bundle action.

File: .github/workflows/make_bundle.yml (hunk 1 above)
Comment (Czaki) [inline .github/workflows/make_bundle.yml]:
```suggestion
          sudo ln -s /usr/lib/x86_64-linux-gnu/libffi.so.7 /usr/lib/x86_64-linux-gnu/libffi.so.6
```

Comment (Czaki) [reply .github/workflows/make_bundle.yml]: This problem was already reported here: #3487.

This change is to keep the bundle job working, but maybe it is time to remove briefcase bundles. At least the Linux one, if no one would like to bump the briefcase version. 
https://github.com/napari/napari/blob/0e08cb1cfede6d12fdad0c6aacefd4d635fb5ff8/setup.cfg#L145

**Comment by Czaki** [comment]
> For reference, here's the official issue on the deprecation of this runner image: [actions/runner-images#6002](https://github.com/actions/runner-images/issues/6002)

So we land in this period: February 21, 10.00 UTC – February 21, 22.00 UTC



> I'm a bit surprised that Ubuntu 18.04 is being dropped already - there aren't any other images we could use right?

We could set up jobs using docker containers on a newer version of ubuntu on the host system.

**Comment by psobolewskiPhD** [comment]
Looking at: https://ubuntu.com/about/release-cycle#ubuntu
ubuntu 18.04 was released in april 2018, which would be a bit before python 3.7 if I'm following correctly. It's being dropped from LTS in april 2023.
I'm not a linux user, but it seems if we follow the general python and numpy guidelines we may as well follow this one too?

**Comment by Czaki** [comment]
> ubuntu 18.04 was released in april 2018, which would be a bit before python 3.7 if I'm following correctly. It's being dropped from LTS in april 2023.

In general, yes. There is ESM support, but I do not this should be important for us.

> I'm not a linux user, but it seems if we follow the general python and numpy guidelines we may as well follow this one too?

From time to time, people worry about updating the system because of some old (propertiary) software. Like people stuck on macOS 10.14 because of 32-bits code.  

But again, I'm not sure if we should worry until someone would like to donate time/money for that purpose.

**Comment by psobolewskiPhD** [comment]
> I'm not sure if we should worry until someone would like to donate time/money for that purpose.

Agreed. I think we should follow the general trends. It's likely napari will keep working, but if it's not supported, we shouldn't dwell on it. If someone reports a bug, we can take it on a case by case basis.

**Review by psobolewskiPhD — APPROVED** [review]

**Review by andy-sweet — APPROVED** [review]
Approving because the version bump looks good, and because the symlinks are not too bad and are for briefcase which may be on the way out.

**Comment by Czaki** [comment]
I mark is as ready for merge, but do not perform a fast merge as workflow should work now.

**Comment by andy-sweet** [comment]
I think we hit another temporary outage period: https://github.com/actions/runner-images/issues/6002

@Czaki: Is there a reason you haven't merged this yet? We'll hit an temporary outage period every week until April 1st (not the best date GitHub...) when it becomes permanent.

**Comment by Czaki** [comment]
> @Czaki: Is there a reason you haven't merged this yet? We'll hit an temporary outage period every week until April 1st (not the best date GitHub...) when it becomes permanent.

I forgot about this.

**Comment by Czaki** [comment]
> @Czaki: Is there a reason you haven't merged this yet? We'll hit an temporary outage period every week until April 1st (not the best date GitHub...) when it becomes permanent.

I forgot about this.

**Pull request merged**